modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItDataTypesTest.java

```diff
@@ -86,6 +87,23 @@ public void testUnicodeStrings() {
         }
     }
 
+    /** Test correctness of char types limitation. */
+    @Test
+    public void charLimitation() {
+        sql("create table limitedChar (pk int primary key, f1 VARCHAR(5), f2 CHAR(4))");
+        assertThrows(CalciteContextException.class, () -> sql("insert into limitedChar(pk, f2) values (1, 'aaaadddd')"));
```

Comment:
The exception has message `Cannot assign to target field 'F2' of type VARCHAR(4) from source field ''aaaadddd'' of type VARCHAR(4)`. Something definitely is wrong here...

Reply:
this one even better: `Cannot assign to target field 'F1' of type VARCHAR(2) from source field ''a  ' :: CHAR' of type VARCHAR(2)`